### PR TITLE
nxp: patches: linux-imx: Read ACK data to host boot the ADSD3500 SPI/I2C

### PR DESCRIPTION
**Pull request by SPadmana95**
ADSD3500 SPI/I2C linux driver patches to support the read acknowledgement data on every header and payload packet of ADSD3500 host boot firmware

### CODE REVIEW

File: sdcard-images-utils/nxp/patches/linux-imx/0063-drivers-media-spi-adsd3500-spi.c-Read-ACK-data-after.patch [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
++	uint8_t read_val[4];
++	int ret;
++
++	ret = adsd3500_spi_read_response(client, read_val, sizeof(read_val));
++	if (ret < 0) {
++		dev_err(adsd3500->dev, "Read ACK cmd failed.\n");
++		return -EIO;
++	}
++	else{
++		if(read_val[0] != 0x0B){
```
Comment (andrestraker) [inline sdcard-images-utils/nxp/patches/linux-imx/0063-drivers-media-spi-adsd3500-spi.c-Read-ACK-data-after.patch:34]: * Add a space before the '(' and after the ')'.
* Add a comment on the error "!= 0x0B" means.

File: sdcard-images-utils/nxp/patches/linux-imx/0063-drivers-media-spi-adsd3500-spi.c-Read-ACK-data-after.patch [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ 	struct adsd3500 *adsd3500 = to_adsd3500(sd);
+@@ -1546,9 +1570,18 @@ static int adsd3500_send_host_boot_data(struct v4l2_subdev *sd, uint8_t* data, i
+ 		dev_dbg(adsd3500->dev, "Send Header data\n");
+ 		ret = spi_write(client, header_packet, HEADER_SIZE_IN_BYTES + 1);
+ 		if (ret < 0) {
+-			dev_err(adsd3500->dev, "Failed to write the header packet of segment %d\n", curr_segment);
++			dev_err(adsd3500->dev, "Failed to write the header packet of segment %d\n", curr_segment + 1);
+ 			return -EIO;
+ 		}
++		if(header_packet[04] == 0x55){
```
Comment (andrestraker) [inline sdcard-images-utils/nxp/patches/linux-imx/0063-drivers-media-spi-adsd3500-spi.c-Read-ACK-data-after.patch:55]: * Add a space before the '(' and after the ')'.
* Add a comment on the error "!= 0x55" means.

File: sdcard-images-utils/nxp/patches/linux-imx/0064-drivers-media-i2c-adsd3500.c-Acknowledgement-read-da.patch [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ 	struct adsd3500 *adsd3500 = to_adsd3500(sd);
+@@ -1350,9 +1374,18 @@ static int adsd3500_send_host_boot_data(struct v4l2_subdev *sd, uint8_t* data, i
+ 		dev_dbg(adsd3500->dev, "Send Header data\n");
+ 		ret = i2c_master_send(client, header_packet, HEADER_SIZE_IN_BYTES);
+ 		if (ret < 0) {
+-			dev_err(adsd3500->dev, "Failed to write the header packet of segment %d\n", curr_segment);
++			dev_err(adsd3500->dev, "Failed to write the header packet of segment %d\n", curr_segment + 1);
+ 			return -EIO;
+ 		}
++		if(header_packet[03] == 0x55){
```
Comment (andrestraker) [inline sdcard-images-utils/nxp/patches/linux-imx/0064-drivers-media-i2c-adsd3500.c-Acknowledgement-read-da.patch:55]: * Add a space before the '(' and after the ')'.
* Add a comment on the error "!= 0x55" means.

File: sdcard-images-utils/nxp/patches/linux-imx/0064-drivers-media-i2c-adsd3500.c-Acknowledgement-read-da.patch [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
++			break;
++		}
++		ret = adsd3500_read_ack(sd);
++		if(ret < 0) {
++			dev_err(adsd3500->dev, "Failed to read the acknowledgement header packet segment: %d\n", curr_segment + 1);
++			return -ENXIO;
++		}
+ 		msleep(5);
+ 
+ 		if(segment_size[curr_segment] != 0){
```
Comment (andrestraker) [inline sdcard-images-utils/nxp/patches/linux-imx/0064-drivers-media-i2c-adsd3500.c-Acknowledgement-read-da.patch:66]: * Add a space before the '(' and after the ')'.

**Comment by SPadmana95** [comment]
Spaces were added in the conditional statement and comments for opcode comparison. The ADSD3500 SPI and I2C Linux driver source files have been added.

**Review by andrestraker — APPROVED** [review]

**Pull request merged**